libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.h

```diff
@@ -29,16 +30,22 @@ class AP_BattMonitor_UAVCAN : public AP_BattMonitor_Backend
 
     bool has_temperature() const override { return _has_temperature; }
 
-    bool has_current() const override {
-        return true;
-    }
+    bool has_current() const override { return true; }
+
+    bool has_time_remaining() const override { return _has_time_remaining; }
+
+    bool has_cell_voltages() const override { return _has_cell_voltages; }
+
+    bool get_cycle_count(uint16_t &cycles) const override;
 
```

Comment:
You should add somewhere around here something like this so that the consumed_energy information can be sent to the ground stations.
```suggestion
    bool has_consumed_energy() const { return _has_consumed_energy; }
```